janusgraph-core/src/main/java/org/janusgraph/graphdb/management/ConfigurationManagementGraph.java

```diff
@@ -0,0 +1,310 @@
+        createIndexIfDoesNotExist(TEMPLATE_INDEX, PROPERTY_TEMPLATE, Boolean.class, false);
+        createIndexIfDoesNotExist(PROPERTY_CREATED_USING_TEMPLATE, PROPERTY_CREATED_USING_TEMPLATE, Boolean.class, false);
+    }
+
+    private synchronized void initialize() {
+        if (null != instance) {
+            final String errMsg = "ConfigurationManagementGraph should be instantiated just once, by the JanusGraphManager.";
+            throw new ConfigurationManagementGraphAlreadyInstantiatedException(errMsg);
+        }
+        this.instance = this;
+    }
+
+    /**
+     * If one of your "graphs" key was equivalent to \"JanusConfigurationGraph\" in your
+     * YAML file supplied at server start, then we return the ConfigurationManagementGraph
+     * Singleton-- otherwise we throw a {@link ConfigurationManagementGraphNotEnabledException} exception.
+     */
+    public static ConfigurationManagementGraph getInstance() throws ConfigurationManagementGraphNotEnabledException {
+        if (null == instance) {
+            throw new ConfigurationManagementGraphNotEnabledException(
+                "Please add a key named \"JanusConfigurationGraph\" to the \"graphs\" property " +
+                "in your YAML file and restart the server to be able to use the functionality " +
+                "of the ConfigurationManagementGraph class."
+            );
+        }
+
+        return instance;
+    }
+
+    /**
+     * Create a configuration according to the supplied {@link MapConfiguration}; you must include
+     * the property "graph.graphname" with a {@link String} value in the configuration; you can then
+     * open your {@link JanusGraph} using {@link String} graph.graphname without having to supply the
+     * a Configuration or File each time using the {@ConfiguredGraphFactory}.
+     */
+    public void createConfiguration(final MapConfiguration config) {
+        Preconditions.checkArgument(config.containsKey(PROPERTY_GRAPH_NAME),
+                                    String.format("Please include the property \"%s\" in your configuration.",
+                                                  PROPERTY_GRAPH_NAME
+                                    ));
+        final Map<String, Object> map = config.getMap();
+        final Vertex v = graph.addVertex(T.label, VERTEX_LABEL);
+        map.forEach(v::property);
+        v.property(PROPERTY_TEMPLATE, false);
+        graph.tx().commit();
+    }
+
+    /**
+     * Create a "Template_Configuration" according to the supplied {@link MapConfiguration}; if
+     * you already created a Template_Configuration or the supplied {@link MapConfiguration}
+     * contains the property "graph.graphname", we throw a {@link RuntimeException}; you can then use
+     * this Template_Configuration to create a {@JanusGraph} using the {@link
+     * ConfiguredGraphFactory} create signature and supplying a new {@link String} graphName.
```

Comment:
Change `{JanusGraph}` to just "graph" or else use `@link` and import/use fully-qualified name. Likewise for ConfiguredGraphFactory either import or use fully-qualified name.